File: Problems/1491. (E) Average Salary Excluding the Minimum and Maximum Salary/Java/solution.java
class Solution {
  public double average(int[] salary) {
    // iteration

    int s_min = 100001;
    int s_max = 999;
    int total = 0;
    for (int s: salary) {
      s_min = Math.min(s_min, s);
      s_max = Math.max(s_max, s);
      total += s;
    }
    return (double) (total - s_min - s_max) / (salary.length - 2);
  }
}

File: Problems/1491. (E) Average Salary Excluding the Minimum and Maximum Salary/Python3/solution.py
class Solution:
  def average(self, salary: List[int]) -> float:
    # iteration
    
    s_min, s_max, total = 10**6+1, 999, 0
    for s in salary:
      s_min = min(s_min, s)
      s_max = max(s_max, s)
      total += s
    
    return (total - s_min - s_max) / (len(salary) - 2)

